apis/projectcontour/v1alpha1/contourconfig.go

```diff
@@ -348,6 +348,12 @@ type EnvoyListenerConfig struct {
 	// +optional
 	UseProxyProto *bool `json:"useProxyProtocol,omitempty"`
 
+	// Compression selects the compression type applied in the compression HTTP filter of the default Listener filters.
+	// Values: `gzip` (default), `brotli`, `zstd`, `disabled`.
+	// Setting this to `disabled` will make Envoy skip "Accept-Encoding: gzip,deflate" request header and always return uncompressed response
+	// +optional
+	Compression EnvoyCompressionType `json:"compression,omitempty"`
```

Comment:
I did think (briefly) about making the configurability greater, and/or having a wider range of compression algorithms. In the end I decided against it, at least for this PR - 

- while one could add more compression algorithms I thought that exposing the ones from `envoy/extensions` was a nice and easily explained choice. There are a couple of others in `contrib/envoy/extensions` but gzip/br/zstd seems like the ones that are most likely to be widely wanted. And I certainly didn't want to start thinking about making the algorithm choice be more "pluggable"!
- as for configurability of algorithms I thought it wasn't something I felt I wanted to start making decisions on - I expect contour project members would perhaps not want to expose all the configuration options for every algorithm, but then this leaves you with the question of how to sensibly define some configuration that supports enough configurability for the different algorithms. It felt like a bit of a rabbit hole and I thought it would be something kept separate from this PR - better to try to keep changes as small as possible, and this one is meaty enough already.

Believe it or not I actually took a bit of inspiration from the Contour Philosophy 😆  (I did read around on how to contribute), particularly the bit on [Sensible Defaults](https://projectcontour.io/resources/philosophy/#sensible-defaults). I felt these kind of gave permission to simply offer a choice of a few widely used algorithms, using default settings. While a strategy field could be added here, it would be something of a no-op feature unless we took the time to implement at least basic configurability in it. I feel this would be best left to another PR. Would that be OK?

Reply:
Also now that I think of it, the feature here of most interest to us in our project is the compression disabling - I don't feel I could justify the amount of time it would take to add configurability to the algorithms. Like everyone else we're very busy, and I ended up having to do most of this work in my own time in the evenings and weekends :-(

Reply:
To be transparent I dont want to design how we extend it but I am trying to think through if we leave enough space to extend the api.

>  In the end I decided against it, at least for this PR 

My only issue is that if we go with the approach here and in 3 months we wanted to add another `compression` knob then we have two options:

* Make a breaking change which might be fine due to the fact that this is a `v1alpha1` package.
* Add the option  topline object in `type EnvoyListenerConfig struct`

If we think that these are fine options I think the api is good. I am just proponent of adding another `struct` that has a single member so we can extend just that `struct`

>  Like everyone else we're very busy, and I have ended up having to do most of this work in my own time in the evenings and weekends :-(

Yeah, a lot of the people contributing to this project to this into this capacity so I feel you.

Reply:
I will defer to @projectcontour/maintainers

Reply:
Ah now that you have laid it out like that I see what you mean. Let me take a look at it, hopefully this wouldn't be too large a change.

Reply:
hi @davinci26 I have taken a stab at this in [5dcdc33](https://github.com/projectcontour/contour/pull/6546/commits/5dcdc33aeb0d1b27c24527e34734857ea2583612), could you have a look and see what you think?